.github/workflows/release.yml

```diff
@@ -2,98 +2,124 @@ name: Build & Release (manual)
 
 # Manually triggered
 # $ gh workflow run release.yml
-on: workflow_dispatch
-
-# Builds four frameworks in parallel, then commits checksums
-# releases with the frameworks and checksums attached to the release.
+on:
+  workflow_dispatch:
+    inputs:
+      version:
+        description: Select Release type - major, minor or patch(for version upgrade).
+        required: true
+        default: patch
+        type: choice
+        options:
+        - patch
+        - minor
+        - major
+        
 jobs:
-  build:
+  static-analysis:
```

Comment:
for the static analysis phase, should we do some sort of check here instead of just running it? I know we have a couple known flags from it so it might not make sense to block on it yet.